pkg/cmd/extension/command.go

```diff
@@ -322,7 +338,15 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
 						if err != nil {
 							return err
 						}
-						return m.InstallLocal(wd)
+
+						err = m.InstallLocal(wd)
+						if errors.Is(err, ErrExtensionExecutableNotFound) {
+							if io.IsStdoutTTY() {
+								fmt.Fprintln(io.ErrOut, err.Error())
+							}
+							return nil
```

Comment:
Are there any circumstances where it would be helpful to print this error if not TTY? I imagine someone installing an extension as part of a workflow, this error occurring, and them lacking any useful information to debug it. I did a [code search for `gh extension install`](https://github.com/search?q=%22gh+extension+install%22&type=code) and see a couple examples of this on the first results page.
```suggestion
						if errors.Is(err, ErrExtensionExecutableNotFound) {
							fmt.Fprintln(io.ErrOut, err.Error())
						        return nil
						}
```

Reply:
@williammartin [mentioned that](https://github.com/cli/cli/pull/9933#issuecomment-2486418124), and this was his reasoning to prefer having the message only on TTY:

>  I don't feel strongly either way about just the message. Maybe I'd favour just having it on TTY, since realistically, people who are running it in a script probably understand what the semantics are of the command.

---

> I did a [code search for gh extension install](https://github.com/search?q=%22gh+extension+install%22&type=code) and see a couple examples of this on the first results page

I think a better code search is to [search for `gh extension install .`](https://github.com/search?q=%22gh+extension+install+.%22&type=code) instead, since this message can only happen with this syntax to install a local extension.

We could [take the search a bit further as well](https://github.com/search?q=%22gh+extension+install+.%22+-path%3A*.md+AND+-path%3A*.go&type=code) by removing markdown and Go files, focusing on identifying automations where an extension might be installed locally. I removed markdown from the search because we can loosely assume markdown instructions are for user guided programs and would have a TTY. I removed Go files because it doesn't make sense for this to be in Go files aside from forks of `cli/cli`.

And if you want to see actions workflows specifically, look at [this search](https://github.com/search?q=%22gh+extension+install+.%22+-path%3A*.md+AND+-path%3A*.go+AND+path%3A**.github%2Fworkflows%2F*&type=code).

Either way, I think it is worth saying that this error is not actually an error - it's actually a warning. It does not indicate a failure to install the extension; it is a warning that additional steps are required to have the installed extension work. These additional steps already exist, so existing automations and workflows are likely unaffected and don't need to see this new warning message. I suppose the argument is that someone authoring a new automation or workflow might be interested in seeing this new warning message, or than an existing workflow breaks and it might be useful for them to see this new warning message 🤷 

Though, the user/scriot definitely does receive an error message _later_ when they attempt to run an extension that has no executable 😜 

I suppose the last consideration I'll throw into the mix is that we have some level of precedent in `extension.go` for showing informative messages to the user only when there is a TTY (though I can see the argument that our case is different):

https://github.com/cli/cli/blob/9177b22a213318ccc00a9b08fbac1dd14fb3a839/pkg/cmd/extension/command.go#L168-L170

I don't care particularly strongly. Thoughts?

Reply:
Makes sense. That search you've put together is much better than mine and really shows the space I have to level-up using GH code search 😅 Thanks for taking the time to show me this 🙌 

> I suppose the argument is that someone authoring a new automation or workflow might be interested in seeing this new warning message, or than an existing workflow breaks and it might be useful for them to see this new warning message 🤷

I think this is probably the most useful use-case, but that a new author is likely going to be testing installs TTY and not through automation, so they'll get the error message anyway. Let's leave it as it is, then. If we get feedback from folks that they could have benefited from more guidance, we can revisit 👍